README.md
# tungcafe
Tung Cafe Management


CHANGELOG.md
### Changelog

**VERSION 3.0.0** (11-15-2017)
*Note: A big update, there might be errors, let me know, thanks!*

    - Feat: Upgrade to Angular 5
    - Feat: Upgrade @angular/material to 5.0.0-rc0
    - Feat: Upgrade Angular CLI to 1.5.x to support Angular v5
    - Feat: Switch default solution from Webpack to Angular CLI
    - Feat: Update NPM packages to latest version
    - Docs: Update docs


**VERSION 2.4.0** (10-23-2017)

    - Feat: Upgrade @angular/material to 2.0.0-beta.12
    - Feat: Upgrade Angular CLI to 1.4.x
    - Feat: Update NPM packages
    - Refactor: refactor the code

**VERSION 2.3.0** (08-31-2017)

    - Feat: Upgrade Bootstrap to 4.0.0-beta
    - Feat: Update NPM packages

**VERSION 2.2.4** (08-24-2017)

    - Feat: Upgrade Angular CLI to v1.3.x
    - Feat: Add Data Table example
    - Chore: Remove yarn.lock, use default package-lock.json instead
    - Docs: Update docs

**VERSION 2.2.3** (07-20-2017)

    - Feat: Upgrade Angular CLI to v1.2.2
    - Feat: Update NPM packages to latest versions
    - Docs: Update docs

**VERSION 2.2.2** (07-10-2017)

    - Feat: Upgrade NPM to v5, and add package-lock.json
    - Feat: Upgrade @angular/material to beta.8
    - Feat: Upgrade Angular CLI to v1.2.0
    - Feat: Update NPM packages to latest versions
    - Docs: Update docs

**VERSION 2.2.1** (06-22-2017)

    - Feat: Update NPM packages
    - Feat: Upgrade Angular CLI to v1.1.3
    - Docs: Update docs

**VERSION 2.2.0** (06-01-2017)

    - Feat: Upgrade @angular/material to 2.0.0-beta.6
    - Feat: Upgrade @angular/cli to v1.1.0
    - Feat: Trigger ECharts resize on layout changed
    - Feat: Update NPM packages to latest version
    - Refactor: Change indentation to 2 spaces so it's consistent
    - Refactor: Rename "style/libs" to "style/vendors" for consistence
    - Chore: Remove HammerJS for smaller size since it is now optional
    - Docs: Update docs

**VERSION 2.1.1** (05-23-2017)

    - Feat: Update NPM packages to latest version
    - Docs: Update docs

**VERSION 2.1.0** (05-16-2017)

    - Feat: Upgrade @angular/material to beta.5, add new Datepicker
    - Feat: Update NPM packages to latest version, include Angular 4.1.x
    - Refactor: Remove all .DS_Store system files in production
    - Fix: add baseUrl for webpack
    - Docs: Update the docs

**VERSION 2.0.2** (04-25-2017)

    - Feat: Add linter for TypeScript with TSLint
    - Feat: Update NPM packages to latest version
    - Docs: Update docs

**VERSION 2.0.1** (04-17-2017)

    - Feat: Update NPM packages to latest version
    - Feat: Upgrade @angular/material to beta.3 and add new autocomplete example
    - Fix: Update dark theme style
    - Docs: Add guide on how to remove modules

**VERSION 2.0.0** (04-06-2017)

    - Feat: Upgrade to Angular 4
    - Feat: Update NPM packages to latest version
    - Feat: Add AoT (Ahead of Time) support
    - Feat: Lazy load child modules
    - Feat: Add yarn.lock (If you don't use Yarn, simply ingore it)
    - Chore: Move SlimScrollDirective to SharedModule
    - Refactor: Rename AppMainModule to LayoutModule
    - Docs: Update documentation


**VERSION 1.2.4** (03-28-2017)

    - Docs: Update the documentation

**VERSION 1.2.3** (03-23-2017)

    - Docs: Update the guide on using Angular CLI, make it work, thanks @LeFerdi
    - Feat: Update NPM packages to latest version

**VERSION 1.2.2** (03-19-2017)

    - Docs: Add guide on using with Angular CLI (WIP)
    - Dev: Replace ts-helpers with tslib
    - Refactor: Replace ‘require’ with ‘import’

**VERSION 1.2.1** (03-06-2017)

    - Feat: Update NPM packages to latest version
    - Chore: Rename "client" folder to "src" (follow the convention)
    - Fix: Fix admin logo not centered when sidenav is collapsed problem
    - Fix: Fix header text color when header color option is light or yellow
    - Dev: Add source map support for Scss/Sass
    - Docs: Update documentation
    - Chore: Clean up a bit

**VERSION 1.2.0** (03-02-2017)

    - Feat: Landing page

**VERSION 1.1.0** (02-17-2017)

    - Feat: Upgrade to Angular material2 to beta2 ( breaking changes from beta.1 [here](https://github.com/angular/material2/releases/tag/2.0.0-beta.2) )
    - Feat: Update NPM packages to latest version
    - Fix: Fix navToggler button not vertically centered style
    - Style: Update dark theme style for some UI elements
    - Feat(Search Overlay): Add auto focus
    - Chore: Clean up

**VERSION 1.0.0**

    - Initial release


material/src/app/shared/echarts.directive.ts
import { Directive, ElementRef, Input, HostListener, AfterViewInit, OnDestroy } from '@angular/core';
import { Subscription } from 'rxjs/Subscription';
import * as echarts from 'echarts';
import 'echarts/theme/macarons';
import { LayoutService } from '../layout/layout.service';

@Directive({ selector: '[myECharts]' })

export class EChartsDirective implements AfterViewInit, OnDestroy {
  el: ElementRef;
  subscription: Subscription;
  constructor(el: ElementRef, private layoutService: LayoutService) {
    this.el = el;
    this.subscription = layoutService.echartsState$.subscribe((state) => {
      this.resizeChart(state);
    });
  }

  @Input() EChartsOptions: any;
  private myChart;
  private timer;

  ngAfterViewInit() {
    this.myChart = echarts.init(this.el.nativeElement, 'macarons');
    if (!this.EChartsOptions) { return; }

    this.myChart.setOption(this.EChartsOptions);
  }

  ngOnDestroy() {
    if (this.myChart) {
      this.myChart.dispose();
      this.myChart = null; // https://bitbucket.org/iarouse/angular-material/commits/5eec2667b5496edfa1cc0896333b83e188a35676
    }
  }

  resizeChart = (state) => {
    // console.log('state: ' + state)
    setTimeout(() => { // Important
      if (this.myChart) {
        this.myChart.resize();
      }
    }, 300)
  }

  @HostListener('window:resize')
  onResize() {
    this.resizeChart(true);
  }
}


src/app/config.ts

function makeAppConfig() {
  const date = new Date();
  const year = date.getFullYear();

  const AppConfig = {
    brand: 'TÙNG CAFE',
    user: 'Tùng Đinh',
    year,
    layoutBoxed: false,               // true, false
    navCollapsed: false,              // true, false
    navBehind: false,                 // true, false
    fixedHeader: true,                // true, false
    sidebarWidth: 'middle',           // small, middle, large
    theme: 'light',                   // light, gray, dark
    colorOption: '34',                // 11,12,13,14,15,16; 21,22,23,24,25,26; 31,32,33,34,35,36
    AutoCloseMobileNav: true,         // true, false. Automatically close sidenav on route change (Mobile only)
    productLink: '/'
  };

  return AppConfig;
}

export const APPCONFIG = makeAppConfig();
